src/client/address_book/impl.h

```diff
@@ -59,6 +59,42 @@ namespace kovri {
 namespace client {
 
 class AddressBookSubscriber;
+/// @class BookEntry
+/// @brief Container for Address Book entry
+class BookEntry
+{
+ public:
+  /// @brief Create address book entry from a hostname & identity hash
+  /// @param host Kovri hostname
+  /// @param address Kovri identity hash
+  /// @throw std::invalid_argument on empty hostname
+  BookEntry(const std::string& host, const kovri::core::IdentHash& address);
+
+  /// @brief Create address book entry from a host & base64-encoded address
+  /// @param host Kovri hostname
+  /// @param address Base64-encoded Kovri address
+  /// @throw std::invalid_argument on empty hostname
+  /// @throw std::invalid_argument on malformed address
```

Comment:
Malformed, in this case, is not considered std::invalid_argument because the argument is valid as it contains data - but the data itself isn't valid.